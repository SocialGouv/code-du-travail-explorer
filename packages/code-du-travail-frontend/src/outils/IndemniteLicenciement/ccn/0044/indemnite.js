import { round } from "../../../common/math";
/**
 * calcul de l'indemnite de licenciement conventionnelle
 * des industries chimiques et connexes
 */

const contains = (arr, value) => arr.indexOf(value) !== -1;

export function getSalaireRef({
  salaireRefLegal
  // groupe,
  // salaires,
  // salaire,
  // dateNotification
}) {
  return salaireRefLegal;
}

export function getIndemnite({
  salaireRef,
  indemnite,
  anciennete,
  age,
  hasOpe = false,
  isEco = false,
  groupe = "I"
}) {
  // ancienneté en année
  const anneeAncienete = Math.floor(anciennete);
  let indemniteConventionnelle = 0;
  let formuleConventionnelle = "";

  // l'entreprise n'est pas affilié a une Organisation Patronale Employeur

  // le salarié appartient au groupe I, II, III
  if (contains(["I", "II", "III"], groupe)) {
    // le salarié a entre 1 et 2ans d'anciennté
    if (anneeAncienete >= 1 && anneeAncienete < 2) {
      indemniteConventionnelle = indemnite;
      // dans le cas ou il y a un OPE et un licenciement eco
      if (hasOpe && isEco) {
        indemniteConventionnelle = salaireRef;
        formuleConventionnelle = `${round(salaireRef)}`;
      }
    }
    // le salarié a plus de 2ans d'anciennté
    else if (anneeAncienete >= 2) {
      indemniteConventionnelle = (3 / 10) * salaireRef * anciennete;
      formuleConventionnelle = `3/10 * ${round(salaireRef)} * ${round(
        anciennete
      )} `;
    }
    // Pour les salarié avec plus de 5ans d'ancienneté
    if (anneeAncienete >= 5) {
      // le salarié a entre 50 et 55ans
      if (age >= 50 && age < 55) {
        indemniteConventionnelle += salaireRef;
        formuleConventionnelle += ` + ${round(salaireRef)}`;
        if (isEco && hasOpe) {
          indemniteConventionnelle += salaireRef;
          formuleConventionnelle += ` * 2`;
        }
      }
      // le salarié a plus de 55ans
      else if (age >= 55) {
        indemniteConventionnelle += salaireRef * 2;
        formuleConventionnelle += ` + ${round(salaireRef)} * 2`;
      }
    }
    // l'indemnité ne peut exéder 14mois de salaire
    if (indemniteConventionnelle > salaireRef * 14) {
      indemniteConventionnelle = salaireRef * 14;
      formuleConventionnelle = `\\min ${round(salaireRef)} * 14`;
    }
  }
  // le salarié appartient au groupe IV
  else if (groupe === "IV") {
    // le salarié a entre 1 et 2ans d'anciennté
    if (anneeAncienete >= 1 && anneeAncienete < 2) {
      indemniteConventionnelle = indemnite;
      // dans le cas ou il y a un OPE et un licenciement eco
      if (hasOpe && isEco) {
        indemniteConventionnelle = salaireRef;
        formuleConventionnelle = `${round(salaireRef)}`;
      }
    }
    // le salarié a entre 2 et 10ans d'anciennté
    else if (anneeAncienete >= 2 && anneeAncienete < 10) {
      indemniteConventionnelle = (3 / 10) * salaireRef * anciennete;
      formuleConventionnelle = `3 / 10 * (${round(salaireRef)} *  (${round(
        anciennete
      )} / 12)`;
    }
    // le salarié a entre 10 et 20ans d'anciennté
    else if (anneeAncienete >= 10 && anneeAncienete < 20) {
      indemniteConventionnelle = (4 / 10) * salaireRef * anciennete;
      formuleConventionnelle = `(4 / 10) * ${round(salaireRef)} * (${round(
        anciennete
      )}`;
    }
    // le salarié a plus de 20ans d'anciennté
    else if (anneeAncienete >= 20) {
      indemniteConventionnelle = (5 / 10) * salaireRef * anciennete;
      formuleConventionnelle = `(5 / 10) *  ${round(salaireRef)} * (${round(
        anciennete
      )}`;
    }
    // Pour les salariés qui ont plus de 5ans d'ancienneté
    if (anneeAncienete >= 5) {
      // le salarié a entre 50 et 55ans
      if (age >= 50 && age < 55) {
        indemniteConventionnelle += salaireRef;
        formuleConventionnelle += `+ ${round(salaireRef)}`;
        if (hasOpe && isEco) {
          indemniteConventionnelle += salaireRef;
          formuleConventionnelle += `* 2`;
        }
      }
      // le salarié a entre 50 et 55ans
      else if (age >= 55) {
        indemniteConventionnelle += salaireRef * 2;
        formuleConventionnelle += `+ ${round(salaireRef)} * 2`;
      }
    }
    // l'indemnité ne peut exéder 18mois de salaire
    if (indemniteConventionnelle > salaireRef * 18) {
      indemniteConventionnelle = salaireRef * 18;
      formuleConventionnelle = `\\min (${round(
        salaireRef
      )} * 18, ${formuleConventionnelle})`;
    }
  }
  // le salarié appartient au groupe V
  else if (groupe === "V") {
    // le salarié a entre 1 et 2ans d'anciennté
    if (anneeAncienete >= 1 && anneeAncienete < 2) {
      indemniteConventionnelle = indemnite;
      if (hasOpe && isEco) {
        indemniteConventionnelle = salaireRef;
        formuleConventionnelle = `${round(salaireRef)}`;
      }
    }
    // le salarié a entre 2 et 10ans d'anciennté
    else if (anneeAncienete >= 2 && anneeAncienete < 10) {
      indemniteConventionnelle = (4 / 10) * salaireRef * anciennete;
      formuleConventionnelle = `(4 / 10) * ${round(salaireRef)}  * ${round(
        anciennete
      )}`;
    }
    // le salarié a entre 10 et 15ans d'anciennté
    else if (anneeAncienete >= 10 && anneeAncienete < 15) {
      indemniteConventionnelle = (6 / 10) * salaireRef * anciennete;
      formuleConventionnelle = `(6 / 10) * ${round(salaireRef)}  * ${round(
        anciennete
      )}`;
    }
    // le salarié a plus de 15ans d'anciennté
    else if (anneeAncienete >= 15) {
      indemniteConventionnelle = (8 / 10) * salaireRef * anciennete;
      formuleConventionnelle = `(8 / 10) * ${round(salaireRef)}  * ${round(
        anciennete
      )}`;
    }
    // Pour les salariés qui ont plus de 5ans d'ancienneté
    if (anneeAncienete > 6) {
      // le salarié a entre 45 et 55ans
      if (age >= 45 && age < 55) {
        indemniteConventionnelle += salaireRef;
        formuleConventionnelle += ` + ${round(salaireRef)}`;
        if (hasOpe && isEco && age >= 50) {
          indemniteConventionnelle += salaireRef;
          formuleConventionnelle += `* 2`;
        }
      }
      // le salarié a plus de 55ans
      else if (age >= 55) {
        indemniteConventionnelle += salaireRef * 2;
        formuleConventionnelle += `${round(salaireRef)} * 2`;
      }
    }
    // l'indemnité ne peut exéder 20mois de salaire
    if (indemniteConventionnelle > salaireRef * 20) {
      formuleConventionnelle = `\\min (${round(
        salaireRef
      )} * 20, ${formuleConventionnelle})`;
      indemniteConventionnelle = salaireRef * 20;
    }
  }

  if (
    isEco &&
    hasOpe &&
    anneeAncienete >= 2 &&
    indemniteConventionnelle < salaireRef * 2
  ) {
    indemniteConventionnelle = salaireRef * 2;
    formuleConventionnelle = `${round(salaireRef)} * 2`;
  }

  return {
    indemniteConventionnelle: round(indemniteConventionnelle),
    formuleConventionnelle
  };
}
