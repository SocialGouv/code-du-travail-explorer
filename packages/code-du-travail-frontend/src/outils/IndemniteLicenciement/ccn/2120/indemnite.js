import { sum, round } from "../../../common/math";
import { parse } from "../../../common/date";
import { max, min, isAfter, differenceInMonths } from "date-fns";
import {
  CADRE,
  NON_CADRE,
  NE_SAIT_PAS,
  DISCIPLINAIRE,
  NON_DISCIPLINAIRE,
  ECONOMIQUE,
  optionCategorie,
  optionMotifs,
} from "./Step";

function getSalaireRef({
  hasTempsPartiel = false,
  hasSameSalaire = false,
  salaire,
  salairePeriods,
  salaires = [],
  anciennete,
}) {
  const salaryValues = salaires.map((a) => a.salary);

  // calcul du salaire de reference
  if (hasTempsPartiel) {
    return salairePeriods.reduce(
      (salaire, period) =>
        salaire +
        (parseInt(period.salary, 10) * parseInt(period.duration, 10)) /
          12 /
          anciennete,
      0
    );
  }
  return hasSameSalaire ? (12 / 13) * salaire : (1 / 13) * sum(salaryValues);
}

function getIndemnite({
  salaireRef,
  dateEntree,
  dateSortie,
  anciennete,
  categorie = NON_CADRE,
  motif,
}) {
  const dEntree = parse(dateEntree);
  const dSortie = parse(dateSortie);

  let indemniteConventionnelle = 0;
  let formula = "";
  let error;
  const labels = {
    "Salaire de référence (Sref)": round(salaireRef),
    "Motif du licenciement": optionMotifs[motif],
    Catégorie: optionCategorie[categorie],
  };
  if (anciennete < 1) {
    return {
      indemniteConventionnelle,
      infoCalculConventionnel: {
        formula: "-",
        labels,
      },
      error:
        "La convention collective prévoit une indemnité conventionnelle de licenciement à partir d'un an d'ancienneté",
    };
  }
  if (motif === DISCIPLINAIRE) {
    return {
      indemniteConventionnelle,
      infoCalculConventionnel: {
        formula: "-",
        labels,
      },
      error:
        "La convention collective prévoit le droit à l’indemnité légale en cas de licenciement pour motif disciplinaire, sauf pour faute grave ou lourde",
    };
  }
  const year2002 = new Date("2002-01-01");

  const nbSemestreAvant2002 = Math.floor(
    differenceInMonths(year2002, min([year2002, dEntree])) / 6
  );
  const nbSemestreApres2002 = Math.floor(
    differenceInMonths(dSortie, max([dEntree, year2002])) / 6
  );

  labels["Nombre de semestres avant 2002 (S1)"] = round(nbSemestreAvant2002);
  labels["Nombre de semestres après 2002 (S2)"] = round(nbSemestreApres2002);

  if (motif === NON_DISCIPLINAIRE) {
    if (nbSemestreAvant2002 > 0) {
      indemniteConventionnelle =
        (1 / 2) * (13 / 14.5) * salaireRef * nbSemestreAvant2002;
      formula = `1 / 2 * 13 /14.5  * Sref * S1 + `;
    }

    indemniteConventionnelle += (1 / 5) * salaireRef * nbSemestreApres2002;
    formula += `1 / 5 * Sref * S2`;
  } else if (motif === ECONOMIQUE) {
    if (nbSemestreAvant2002 > 0) {
      indemniteConventionnelle = (1 / 2) * salaireRef * nbSemestreAvant2002;
      formula = `1 / 2 * Sref * S1 + `;
    }
    indemniteConventionnelle += (1 / 4) * salaireRef * nbSemestreApres2002;
    formula += `1 / 4 * Sref * S2`;
  }

  const isEmbaucheAfter1999 = isAfter(dEntree, new Date("1999-12-31"));

  if (isEmbaucheAfter1999) {
    const plafond = {
      [CADRE]: 24,
      [NON_CADRE]: 18,
      [NE_SAIT_PAS]: 18,
    };

    if (
      indemniteConventionnelle >
      plafond[categorie] * (13 / 14.5) * salaireRef
    ) {
      indemniteConventionnelle = plafond[categorie] * (13 / 14.5) * salaireRef;
      formula = `${plafond[categorie]} * 13 / 14.5 * Sref)`;
    }
  } else {
    const plafond = {
      [ECONOMIQUE]: 18,
      [NON_DISCIPLINAIRE]: 15,
    };

    if (indemniteConventionnelle > plafond[motif] * salaireRef) {
      indemniteConventionnelle = plafond[motif] * salaireRef;
      formula = `${plafond[motif]} * Sref)`;
    }
  }

  return {
    indemniteConventionnelle: round(indemniteConventionnelle),
    infoCalculConventionnel: { formula, labels },
    error,
  };
}
export { getSalaireRef, getIndemnite };
