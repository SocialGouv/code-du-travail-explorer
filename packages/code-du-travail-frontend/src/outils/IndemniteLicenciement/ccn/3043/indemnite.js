import { round } from "../../../common/math";
/**
 * calcul de l'indemnite de licenciement conventionnelle
 * PROPRETE ENTREPRISES ET SERVICES ASSOCIES - IDCC 3043
 * https://github.com/SocialGouv/code-du-travail-numerique/issues/883
 */

export function getSalaireRef({ salaireRefLegal }) {
  return salaireRefLegal;
}

export function getIndemnite({ salaireRef, anciennete }) {
  let error;
  let remainingAnciennete = anciennete;
  let indemniteConventionnelle = 0;
  let formulesConventionnelles = [];

  if (remainingAnciennete > 10) {
    indemniteConventionnelle +=
      (1 / 5) * salaireRef * (remainingAnciennete - 10);
    formulesConventionnelles.push(
      `1/5 * ${round(salaireRef)} * (${remainingAnciennete} - 10)`
    );
    remainingAnciennete = 10;
  }
  if (remainingAnciennete > 5) {
    indemniteConventionnelle = (1 / 6) * salaireRef * (remainingAnciennete - 5);
    formulesConventionnelles.push(
      `1/6 * ${round(salaireRef)} * (${remainingAnciennete} - 5)`
    );
    remainingAnciennete = 5;
  }
  if (remainingAnciennete > 2) {
    indemniteConventionnelle = (1 / 10) * salaireRef * remainingAnciennete;
    formulesConventionnelles.push(
      ` + 1/10 * ${round(salaireRef)} * ${remainingAnciennete}`
    );
  } else {
    error =
      "Aucune indemnité de licenciement n'est prévue en deça de 2 ans d'ancienneté.";
  }

  return {
    indemniteConventionnelle: round(indemniteConventionnelle),
    formuleConventionnelle: formulesConventionnelles
      .map(formula => `( ${formula} )`)
      .join(" + "),
    error
  };
}
