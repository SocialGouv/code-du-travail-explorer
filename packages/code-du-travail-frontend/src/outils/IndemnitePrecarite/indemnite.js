import { sum } from "../common/math";

function getIndemnitePrecarite({ salaire, salaires, typeRemuneration }) {
  switch (typeRemuneration) {
    case "mensuel": {
      const sommeSalaires = sum(salaires.map(s => s.salaire));
      return {
        indemnite: (1 / 10) * sommeSalaires,
        formule: `1/10 * somme(salaires)`,
        inputs: { "somme des salaires": sommeSalaires }
      };
    }
    case "total": {
      return {
        indemnite: (1 / 10) * salaire,
        formule: `1/10 * somme(salaires)`,
        inputs: { "somme des salaires": salaire }
      };
    }
    default:
      return { indemnite: 0, formule: "", inputs: {} };
  }
}

export { getIndemnitePrecarite };
