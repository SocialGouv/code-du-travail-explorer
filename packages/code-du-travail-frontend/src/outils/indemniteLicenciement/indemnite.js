/*

calcul de l'indemnite de licenciement ou rupture conventionnelle

https://github.com/SocialGouv/code-du-travail-explorer/issues/39

*/

const sum = arr => arr.reduce((sum, c) => sum + parseFloat(c), 0);

// todo: ensure effectivity is not at day-1 for isR12342
// https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000019225838&cidTexte=LEGITEXT000006072050

const getIndemnite = data => {
  const {
    salaires,
    primes,
    anciennete,
    isR12342,
    fauteGrave,
    calculConvention
  } = data;

  // a) moyenne des douze derniers mois
  // (a)=moyenne((mois1+prime)+(mois2+prime)... (mois 12+prime))
  const moyenneSalaires =
    (sum(salaires) + (primes || 0)) / salaires.length || 0;
  // b) moyenne des 3 derniers mois
  // (b)=moyenne((mois10+prime)+(mois11+prime)... (mois 12+prime))
  const moyenne3DerniersMois =
    (sum(salaires.filter((_, i) => i < 3)) + (primes / 12) * 3) / 3;
  // si b>a, (c)=(b) sinon (c)=a
  const meilleurMoyenne = Math.max(moyenneSalaires, moyenne3DerniersMois);

  let indemnite = 0;
  const isSmallAnciennete = anciennete / 12 <= 10; // 10 years
  if (isR12342) {
    // "calculavant2017" : date du licenciement < 26 / 09 / 2017
    // Si "ancienneté inférieure ou égale à 10 ans
    if (isSmallAnciennete) {
      // indemnite = 1 / 5 * c * 10
      indemnite = ((1 / 5) * meilleurMoyenne * anciennete) / 12;
    } else {
      // Si ancienneté supérieur à 10 ans:
      // indemnite = 1 / 5 * c * 10 + 2 / 5 * c * d
      indemnite =
        (1 / 5) * meilleurMoyenne * 10 +
        (2 / 5) * meilleurMoyenne * (anciennete / 12 - 10);
    }
  } else {
    // Si "ancienneté inférieure ou égale à 10 ans
    if (isSmallAnciennete) {
      // indemnite = 1 / 4 * c * 10
      indemnite = ((1 / 4) * meilleurMoyenne * anciennete) / 12;
    } else {
      // Si ancienneté supérieurd à 10 ans:
      //indemnite = 1 / 4 * c * 10 + 1 / 3 * c * d
      indemnite =
        (1 / 4) * meilleurMoyenne * 10 +
        (1 / 3) * meilleurMoyenne * (anciennete / 12 - 10);
    }
  }

  const errors = [];

  if (anciennete < 12) {
    errors.push({
      type: "warning",
      message:
        "L'indemnité de licenciement est dûe au-delà d'un an d'ancienneté."
    });
    indemnite = 0;
  }

  if (fauteGrave) {
    indemnite = 0;
    errors.push({
      type: "warning",
      message: `L’indemnité légale de licenciement n'est pas dûe en cas de faute grave.
        <br/><br/>
        Vous reporter à la lettre de notification de licenciement, lorsqu'il est invoqué le motif de faute grave doit apparaître précisément dans le courrier.`
    });
  }

  let calculCC;

  if (calculConvention) {
    calculCC = calculConvention({ ...data, indemnite });
  }
  return {
    isSmallAnciennete,
    anciennete,
    isR12342,
    moyenneSalaires,
    moyenne3DerniersMois,
    meilleurMoyenne,
    indemnite,
    errors,
    calculCC
  };
};

export default getIndemnite;
