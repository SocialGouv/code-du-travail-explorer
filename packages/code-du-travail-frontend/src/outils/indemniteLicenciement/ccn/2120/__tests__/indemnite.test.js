import {
  getIndemnite as getIndemniteConventionnelle,
  getSalaireRef as getSalaireRefConventionnel
} from "../indemnite";
import { getIndemnite, getSalaireRef } from "../../../indemnite";
import {
  CADRE,
  NON_CADRE,
  NE_SAIT_PAS,
  DISCIPLINAIRE,
  NON_DISCIPLINAIRE,
  ECONOMIQUE
} from "../Step";

const initialData = {
  salaires: Array.from({ length: 12 }).fill({ salary: 2000 }),
  hasTempsPartiel: false,
  categorie: NE_SAIT_PAS,
  motif: NON_DISCIPLINAIRE,
  dateEntree: "2018-10-01",
  dateSortie: "2019-09-01",
  anciennete: 0.91,
  branche: "2120",
  age: "55"
};

const tests = [
  {
    title: "2k, 12mois d’ancienneté",
    data: {
      ...initialData
    },
    result: 0
  },
  {
    title: "2k, 10ans et 8mois d’ancienneté, motif disciplinaire",
    data: {
      ...initialData,
      dateEntree: "2009-01-01",
      dateSortie: "2019-09-01",
      motif: DISCIPLINAIRE,
      anciennete: 10.667
    },
    result: 0
  },
  {
    title: "2k, 10ans et 8mois d’ancienneté, motif disciplinaire",
    data: {
      ...initialData,
      dateEntree: "2009-01-01",
      dateSortie: "2019-09-01",
      anciennete: 10.667
    },
    result: 7753.84
  },
  {
    title: "2k, 10ans et 8mois d’ancienneté, motif economique",
    data: {
      ...initialData,
      dateEntree: "2009-01-01",
      dateSortie: "2019-09-01",
      anciennete: 10.667,
      motif: ECONOMIQUE
    },
    result: 9692.3
  },
  {
    title: "2k, 19ans et 8mois d’ancienneté",
    data: {
      ...initialData,
      dateEntree: "2000-01-01",
      dateSortie: "2019-09-01",
      anciennete: 19.667
    },
    result: 16233.42
  },
  {
    title: "2k, 19ans et 8mois d’ancienneté, motif economique",
    data: {
      ...initialData,
      dateEntree: "2000-01-01",
      dateSortie: "2019-09-01",
      anciennete: 19.667,
      motif: ECONOMIQUE
    },
    result: 19846.15
  },
  {
    title: "2k, 20ans et 8mois d’ancienneté, cadre, embauche avant le 2000",
    data: {
      ...initialData,
      dateEntree: "1999-01-01",
      dateSortie: "2019-09-01",
      anciennete: 20.667,
      categorie: CADRE
    },
    result: 17888.59
  },
  {
    title:
      "2k, 20ans et 8mois d’ancienneté, motif economique, embauche avant le 2000",
    data: {
      ...initialData,
      dateEntree: "1999-01-01",
      dateSortie: "2019-09-01",
      anciennete: 20.667,
      motif: ECONOMIQUE,
      categorie: CADRE
    },
    result: 21692.3
  },
  {
    title: "2k, 20ans et 8mois d’ancienneté, non cadre, embauche avant le 2000",
    data: {
      ...initialData,
      dateEntree: "1999-01-01",
      dateSortie: "2019-09-01",
      anciennete: 20.667,
      categorie: NON_CADRE
    },
    result: 17888.59
  },
  {
    title:
      "2k, 20ans et 8mois d’ancienneté, ne-sait-pas, embauche avant le 2000",
    data: {
      ...initialData,
      dateEntree: "1999-01-01",
      dateSortie: "2019-09-01",
      anciennete: 20.667,
      categorie: NE_SAIT_PAS
    },
    result: 17888.59
  }
];
describe("getIndemnite", () => {
  tests.forEach(({ title, data, result }) => {
    it(title, () => {
      const salaireRef = getSalaireRefConventionnel({
        ...data
      });
      const res = getIndemniteConventionnelle({
        ...data,
        salaireRef
      });
      expect(res.indemniteConventionnelle).toBe(result);
      expect(res.formula).toMatchSnapshot();
    });
  });

  it("should return an error for anciennete < 2", () => {
    const salaireRef = getSalaireRef({ ...initialData });
    const { indemnite } = getIndemnite({
      salaireRef,
      ...initialData
    });
    const salaireRefConventionnel = getSalaireRefConventionnel({
      ...initialData
    });

    const res = getIndemniteConventionnelle({
      salaireRef: salaireRefConventionnel,
      indemnite,
      ...initialData
    });

    expect(res.error).toMatchInlineSnapshot(
      `"La convention collective prévoit une indemnité conventionnelle de licenciement à partir d’un an d’ancienneté"`
    );
  });
  it("should return an error for motif disciplinaire", () => {
    const salaireRef = getSalaireRef({ ...initialData });
    const { indemnite } = getIndemnite({
      salaireRef,
      ...initialData
    });
    const salaireRefConventionnel = getSalaireRefConventionnel({
      ...initialData
    });

    const res = getIndemniteConventionnelle({
      salaireRef: salaireRefConventionnel,
      indemnite,
      ...initialData,
      dateEntree: "2009-01-01",
      dateSortie: "2019-09-01",
      anciennete: 10.667,
      motif: DISCIPLINAIRE
    });

    expect(res.error).toMatchInlineSnapshot(
      `"La convention collective prévoit le droit à l’indemnité légale en cas de licenciement pour motif disciplinaire, sauf pour faute grave ou lourde"`
    );
  });
});
