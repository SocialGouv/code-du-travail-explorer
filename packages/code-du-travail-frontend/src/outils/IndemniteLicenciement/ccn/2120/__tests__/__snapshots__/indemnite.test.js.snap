// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getIndemnite 2k, 10ans et 8mois d'ancienneté, motif disciplinaire 1`] = `""`;

exports[`getIndemnite 2k, 10ans et 8mois d'ancienneté, motif disciplinaire 2`] = `"1/5 * Sref * S2"`;

exports[`getIndemnite 2k, 10ans et 8mois d'ancienneté, motif economique 1`] = `"1/4 * Sref * S2"`;

exports[`getIndemnite 2k, 12mois d'ancienneté 1`] = `""`;

exports[`getIndemnite 2k, 19ans et 8mois d'ancienneté 1`] = `"1/2 * 13/14.5 * Sref * S1 + 1/5 * Sref * S2"`;

exports[`getIndemnite 2k, 19ans et 8mois d'ancienneté, motif economique 1`] = `"1/2 * Sref * S1 + 1/4 * Sref * S2"`;

exports[`getIndemnite 2k, 20ans et 8mois d'ancienneté, cadre, embauche avant le 2000 1`] = `"1/2 * 13/14.5 * Sref * S1 + 1/5 * Sref * S2"`;

exports[`getIndemnite 2k, 20ans et 8mois d'ancienneté, motif economique, embauche avant le 2000 1`] = `"1/2 * Sref * S1 + 1/4 * Sref * S2"`;

exports[`getIndemnite 2k, 20ans et 8mois d'ancienneté, ne-sait-pas, embauche avant le 2000 1`] = `"1/2 * 13/14.5 * Sref * S1 + 1/5 * Sref * S2"`;

exports[`getIndemnite 2k, 20ans et 8mois d'ancienneté, non cadre, embauche avant le 2000 1`] = `"1/2 * 13/14.5 * Sref * S1 + 1/5 * Sref * S2"`;
